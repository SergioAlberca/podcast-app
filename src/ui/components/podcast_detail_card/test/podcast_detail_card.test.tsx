import { render, screen } from "@/test/test_utils";
import PodcastDetailCard from "../podcast_detail_card";
import { fixtures } from "./fixtures";
import { BrowserRouter } from "react-router-dom";

describe("PodcastCardDetail component tests", () => {
  it("render component", () => {
    const { container } = render(
      <BrowserRouter>
        <PodcastDetailCard podcastDetail={fixtures} />
      </BrowserRouter>
    );
    expect(container).toMatchSnapshot();
  });

  it("Title podcast is in the document", () => {
    render(
      <BrowserRouter>
        <PodcastDetailCard podcastDetail={fixtures} />
      </BrowserRouter>
    );
    expect(screen.getByText(fixtures["im:artist"].label)).toBeInTheDocument();
  });
});
