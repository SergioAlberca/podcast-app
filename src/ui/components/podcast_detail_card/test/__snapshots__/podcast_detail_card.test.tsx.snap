// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PodcastCardDetail component tests > render component 1`] = `
<div>
  <div
    class="_podcast-detail-card_0b225f"
  >
    <a
      href="/podcast/1535809341"
    >
      <img
        alt="podcast_image"
        src="https://is1-ssl.mzstatic.com/image/thumb/Podcasts113/v4/f2/21/fa/f221fabd-017f-5125-633b-f1fe4f39802a/mza_182995249085044287.jpg/170x170bb.png"
      />
    </a>
    <hr />
    <a
      href="/podcast/1535809341"
    >
      <h4>
        The Joe Budden Podcast
      </h4>
      <h6>
        The Joe Budden Network
      </h6>
    </a>
    <hr />
    <p>
      <strong>
        Description:
      </strong>
       
      Tune into Joe Budden and his friends. Follow along the crazy adventures of these very random friends.
    </p>
  </div>
</div>
`;
